### DISCUSSION TITLE
tag helper alternative for Sinatra

### DISCUSSION THREAD

**Discussion by katelovescode** [ElMassimo/vite_ruby#437] [Q&A]
I'm trying to incorporate vite_ruby with my Sinatra app, which as of right now is extremely simple.  I'm including diagrams of the relevant directory tree as well as the contents of the related files below.

I don't believe that `vite_ruby` has tag helpers, so I'm not sure how to implement the tag helpers that are provided for rails (`<%= vite_client_tag %>` and `<%= vite_javascript_tag %>` specifically) manually to include the pipeline into my Sinatra app. 

Any suggestions on the format of the script/client tags if I'm going to include them manually inside the `<head>` tag? 

```
.
├── config/
│   └── vite.json
├── frontend/
│   └── entrypoints/
│       └── application.js
├── views/
│   ├── layout.erb
│   └── index.erb
├── app.rb
├── package.json
└── vite.config.mts
```

```ruby
# app.rb
require 'sinatra'
require 'uri'
require 'net/http'

get '/' do
  uri = URI('https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY')
  res = Net::HTTP.get_response(uri)
  if res.is_a?(Net::HTTPSuccess)
    @content = res.body 
    erb :index
  else
    "No Data was Fetched"
  end
end
```

```js
// vite.config.mts

import { defineConfig } from 'vite'
import RubyPlugin from 'vite-plugin-ruby'

export default defineConfig({
  plugins: [
    RubyPlugin(),
  ],
})
```

```json
{
  "all": {
    "sourceCodeDir": "frontend",
    "watchAdditionalPaths": []
  },
  "development": {
    "autoBuild": true,
    "publicOutputDir": "vite-dev",
    "port": 3036
  },
  "test": {
    "autoBuild": true,
    "publicOutputDir": "vite-test",
    "port": 3037
  }
}
```

```erb
<!-- layout.erb -->

<html>
  <head>
    <title>Example</title>
    <%= vite_client_tag %>  <!-- undefined local method -->
  </head>
  <body>
    <%= yield %>
  </body>
</html>
```

**Comment by ElMassimo** [comment]
Hi Kate!

Unlike the rest of the integrations ([Vite Hanami](https://github.com/ElMassimo/vite_ruby/tree/main/vite_hanami), [Vite Padrino](https://github.com/ElMassimo/vite_ruby/tree/main/vite_padrino)), Sinatra doesn't have a "standard" view layer, it doesn't provide helpers like Padrino does (which is itself an extension around Sinatra).

You can use [these set of helpers as inspiration](https://github.com/ElMassimo/vite_ruby/blob/main/vite_padrino/lib/vite_padrino/tag_helpers.rb). In fact, if `javascript_include_tag`, `stylesheet_link_tag`, and `content_tag` are available as view helpers, you might be able use the [Vite Padrino](https://github.com/ElMassimo/vite_ruby/tree/main/vite_padrino) tag helpers directly.

Let me know what you come up with, could be helpful for other folks using Sinatra.
